### PR TITLE
Rewrited auth scripts

### PR DESCRIPTION
**Pull request by vyskocilpavel**
* All script now timeouts if the script runs too long
* Login functions is now moved to separate helper scripts

### CODE REVIEW

File: proxy_idp_auth_test_script/saml_auth_test_muni_active.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+if [[ ${LAST_URL} ==  "${TEST_SITE}" ]]; then
+    RESULT=$(echo ${HTML} | sed -e 's/.*<body>\s*Result-\(.*\)<.*$/\1/')
+    if [[ $RESULT == "OK " ]]; then
+        end 0 "Successful login"
+    else
+        end 2 "Bad result: ${RESULT}."
+    fi
+else
+    end 2 "Not redirected back to: ${TEST_SITE}."
+fi
```
Comment (dBucik) [inline proxy_idp_auth_test_script/saml_auth_test_muni_active.sh]: Add a newline please

File: proxy_idp_auth_test_active_saml.sh [hunk 2]
```diff
@@ -1,215 +1,65 @@
...
-  fi
+  STATUS=2
+  STATUS_TXT="Unsuccessful login!"
 fi
 
-echo "$status - $statusTxt [MUNI status - $muniStatusTxt(${muniLoginTime}s); CESNET status - $cesnetStatusTxt(${cesnetLoginTime}s)]"
-exit $status
+
+echo "SAML Auth test active - ${STATUS_TXT} [MUNI STATUS - ${MUNI_RESULT}(${MUNI_LOGIN_TIME}s); CESNET STATUS - ${CESNET_RESULT}(${CESNET_LOGIN_TIME}s)] | muni_login_time=${MUNI_LOGIN_TIME} cesnet_login_time=${CESNET_LOGIN_TIME}"
+exit ${STATUS}
```
Comment (dBucik) [inline proxy_idp_auth_test_active_saml.sh]: Missing newline at the end of the file

File: proxy_idp_auth_test_config.sh [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
+PROXY_DOMAIN_NAME=""
 
 # How long is normal for total roundtrip (seconds)
-samlWarningTime=10
-oidcWarningTime=10
+SAML_WARNING_TIME=10
+OIDC_WARNING_TIME=15
+
+# Timeout time
+TIMEOUT_TIME=40
```
Comment (dBucik) [inline proxy_idp_auth_test_config.sh]: Missing newline

File: proxy_idp_auth_test_oidc.sh [hunk 4]
```diff
@@ -1,221 +1,57 @@
...
-    fi
+  STATUS=2
+  STATUS_TXT="Unsuccessful login!"
 fi
 
-echo "$status $basename-$instanceName muni_login_time=$muniLoginTime|cesnet_login_time=$cesnetLoginTime  - $statusTxt [MUNI status - $muniStatusTxt(${muniLoginTime}s); CESNET status - $cesnetStatusTxt(${cesnetLoginTime}s)]"
-exit 0
+
+echo "${STATUS} ${BASENAME}-${INSTANCE_NAME} muni_login_time=${MUNI_LOGIN_TIME}|cesnet_login_time=${CESNET_LOGIN_TIME} ${STATUS_TXT} [MUNI STATUS - ${MUNI_RESULT}(${MUNI_LOGIN_TIME}s); CESNET STATUS - ${CESNET_RESULT}(${CESNET_LOGIN_TIME}s)]"
+exit 0
```
Comment (dBucik) [inline proxy_idp_auth_test_oidc.sh]: missing newline

File: proxy_idp_auth_test_oidc.sh [hunk 5]
```diff
@@ -1,221 +1,57 @@
...
+END_TIME=$(date +%s%N)
+MUNI_TOTAL_TIME=$(expr ${END_TIME} - ${START_TIME})
+MUNI_LOGIN_TIME=$(echo "scale=4;${MUNI_TOTAL_TIME} / 1000000000" | bc -l)
+
+START_TIME=$(date +%s%N)
+CESNET_RESULT=$(timeout ${TIMEOUT_TIME} ${CESNET_LOGIN_CMD})
+CESNET_RC=$?
+END_TIME=$(date +%s%N)
+
+if [ ${CESNET_RC} -gt 2 ]; then
```
Comment (dBucik) [inline proxy_idp_auth_test_oidc.sh:34]: There is a lot of code that is common in different files. Perhaps you could create a separated file with functions to handle the result and just pass it different messages (if they differ of course)?

File: proxy_idp_auth_test_saml.sh [hunk 6]
```diff
@@ -1,207 +1,57 @@
...
+else
+  STATUS=2
+  STATUS_TXT="Unsuccessful login!"
 fi
 
-echo "$status $basename-$instanceName muni_login_time=$muniLoginTime|cesnet_login_time=$cesnetLoginTime $statusTxt [MUNI status - $muniStatusTxt(${muniLoginTime}s); CESNET status - $cesnetStatusTxt(${cesnetLoginTime}s)]"
-exit 0
+
+echo "${STATUS} ${BASENAME}-${INSTANCE_NAME} muni_login_time=${MUNI_LOGIN_TIME}|cesnet_login_time=${CESNET_LOGIN_TIME} ${STATUS_TXT} [MUNI STATUS - ${MUNI_RESULT}(${MUNI_LOGIN_TIME}s); CESNET STATUS - ${CESNET_RESULT}(${CESNET_LOGIN_TIME}s)]"
+exit 0
```
Comment (dBucik) [inline proxy_idp_auth_test_saml.sh]: missing newline

File: proxy_idp_auth_test_script/saml_auth_test_muni.sh [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+if [[ ${LAST_URL} ==  "${TEST_SITE}" ]]; then
+    RESULT=$(echo ${HTML} | sed -e 's/.*<body>\s*Result-\(.*\)<.*$/\1/')
+    if [[ $RESULT == "OK " ]]; then
+        end 0 "Successful login"
+    else
+        end 2 "Bad result: ${RESULT}."
+    fi
+else
+    end 2 "Not redirected back to: ${TEST_SITE}."
+fi
```
Comment (dBucik) [inline proxy_idp_auth_test_script/saml_auth_test_muni.sh]: missing newline

**Review by dBucik — CHANGES_REQUESTED** [review]

**Comment by vyskocilpavel** [comment]
Added missing newlines

**Comment by vyskocilpavel** [comment]
Merging after discussion with Dominik Bučík

**Pull request merged**